test/factorization/par_ilu_kernels.cpp

```diff
@@ -78,6 +78,11 @@ class ParIlu : public CommonTestFixture {
             const auto row_start = row_ptrs[row];
             const auto row_end = row_ptrs[row + 1];
             const int num_row_elements = row_end - row_start;
+            if (num_row_elements == 0) {
+                // nothing to do if the row is empty (avoid UB in the following
+                // initialization of idx_dist)
+                continue;
```

Comment:
I do not get it. creating and destroying something but not using is UB?

Reply:
std::uniform_int_distribution requires end >= start, otherwise e.g. if the computations used a division internally, they might divide by zero, which crashes the program